File: cpp_on_sea/monte_carlo_pi/java/MonteCarloPi.java
import java.util.Random;

public class MonteCarloPi {
  public static void main(String[] args) {
    final long iterations = 10_000_000L;
    long insideCircle = 0;
    Random random = new Random();

    for (long i = 0; i < iterations; i++) {
      double x = random.nextDouble();
      double y = random.nextDouble();
      if (x * x + y * y <= 1.0) {
        insideCircle++;
      }
    }

    double piEstimate = 4.0 * insideCircle / iterations;
    System.out.println("Estimated Pi = " + piEstimate);
  }
}

File: cpp_on_sea/monte_carlo_pi/kotlin/montecarlopi.kt
fun main() {
    // Number of random points to generate
    val iterations = 10_000_000L
    var insideCircle = 0L

    // Random number generator
    val random = java.util.Random()

    // Monte Carlo simulation
    for (i in 1..iterations) {
        val x = random.nextDouble()
        val y = random.nextDouble()
        if (x * x + y * y <= 1.0) {
            insideCircle++
        }
    }

    // Estimate of Pi
    val piEstimate = 4.0 * insideCircle / iterations
    println("Estimated Pi = $piEstimate")
}